flocker/common/runner.py

```diff
@@ -209,45 +250,88 @@ def run_ssh(reactor, username, host, command, **kwargs):
     )
 
 
-def download_file(reactor, username, host, remote_path, local_path):
+DOWNLOAD = object()
+UPLOAD = object()
+
+
+class SCPError(Exception):
+    """
+    An expected ``scp`` error.
     """
-    Run the local ``scp`` command to download a single file from a remote host
-    and kill it if the reactor stops.
 
-    :param reactor: Reactor to use.
-    :param username: The username to use when logging into the remote server.
-    :param host: The hostname or IP address of the remote server.
-    :param FilePath remote_path: The path of the file on the remote host.
-    :param FilePath local_path: The path of the file on the local host.
 
-    :return Deferred: Deferred that fires when the process is ended.  If the
-        file isn't found on the remote server, it fires with ``FileNotFound``.
+class SCPConnectionError(SCPError):
+    """
+    An ``scp`` connection error.
+    """
+
+
+def scp(reactor, username, host, remote_path, local_path,
+        port=22, identity_file=None, direction=DOWNLOAD):
+    """
+    :param reactor: A ``twisted.internet.reactor``.
+    :param bytes username: The SSH username.
+    :param bytes host: The SSH host.
+    :param FilePath remote_path: The path to the remote file.
+    :param FilePath local_path: The path to the local file.
+    :param int port: The SSH TCP port.
+    :param FilePath identity_file: The path to an SSH private key.
+    :param direction: One of ``DOWNLOAD`` or ``UPLOAD``.
```

Comment:
:question: prefer no default direction. Defaulting to either and having the user expect the wrong one could result in them overwriting a file they wanted to copy elsewhere. This seems precisely like the sort of file you don't want overwritten.